text/0015-Service-Reference-Retrieval.md

```diff
@@ -0,0 +1,232 @@
+- Start Date: 2024-10-10
+- CppMicroServices PR: https://github.com/CppMicroServices/CppMicroServices/pull/1044
+
+# Add an free API in that can retrieve a <code>ServiceObject</code>'s <code>ServiceReference</code> from itself
```

Comment:
This specifically talks about `ServiceObject`, but I think more generally, we want to retrieve a `ServiceReference` given a `shared_ptr<T>` service object, where that service object was given to the client by the cppms framework.

Reply:
Hm interesting. I guess I had it in my head that those two things were one and the same, but seemingly not. I'll update